lib/src/logic/questionnaire_controller.dart

```diff
@@ -85,72 +85,99 @@ class QuestionnaireController {
+  }) async {
     List<QuestionnaireItemBundle> itemBundles = [];
+
     try {
-      for (final QuestionnaireItem item in questionnaire.item ?? []) {
+      for (final QuestionnaireItem item in questionnaireItems ?? []) {
         QuestionnaireItemEnableWhenController? enableWhenController =
             getEnableWhenController(item: item, itemBundles: itemBundles);
         final itemType = QuestionnaireItemType.valueOf(item.type.value);
-        final itemView = switch (itemType) {
-          QuestionnaireItemType.string => QuestionnaireStringItemView(
-              item: item,
-              enableWhenController: enableWhenController,
-            ),
-          QuestionnaireItemType.text => QuestionnaireTextItemView(
-              item: item,
-              enableWhenController: enableWhenController,
-            ),
-          QuestionnaireItemType.integer => QuestionnaireIntegerItemView(
-              item: item,
-              enableWhenController: enableWhenController,
-            ),
-          QuestionnaireItemType.decimal => QuestionnaireDecimalItemView(
-              item: item,
-              enableWhenController: enableWhenController,
-            ),
-          QuestionnaireItemType.boolean => QuestionnaireBooleanItemView(
-              item: item,
-              enableWhenController: enableWhenController,
-            ),
-          QuestionnaireItemType.choice => _buildChoiceItemView(
-              item: item, enableWhenController: enableWhenController),
-          QuestionnaireItemType.openChoice => _buildOpenChoiceItemView(
-              item: item, enableWhenController: enableWhenController),
-          QuestionnaireItemType.date ||
-          QuestionnaireItemType.time ||
-          QuestionnaireItemType.dateTime =>
-            QuestionnaireDateTimeItemView(
-              item: item,
-              enableWhenController: enableWhenController,
-              type: DateTimeType.fromQuestionnaireItemType(itemType),
-            ),
-          QuestionnaireItemType.quantity => QuestionnaireQuantityItemView(
-              item: item,
-              enableWhenController: enableWhenController,
-            ),
-          QuestionnaireItemType.url => QuestionnaireUrlItemView(
-              item: item,
-              enableWhenController: enableWhenController,
-            ),
-          QuestionnaireItemType.display => QuestionnaireDisplayItemView(
+
+        if (itemType == QuestionnaireItemType.group) {
+          final childItems = item.item ?? [];
+          final groupTitleItemView = QuestionnaireDisplayItemView(
+            item: item,
+            enableWhenController: enableWhenController,
+          );
+
+          itemBundles.addAll([
+            /// title of the group shown as a DisplayItemView
+            QuestionnaireItemBundle(
               item: item,
-              enableWhenController: enableWhenController,
+              controller: groupTitleItemView.controller,
+              view: groupTitleItemView,
```

Comment:
Not sure about this way of handling the Group item, it looks the same like having the items without groups and just putting a display item before each "group of items", I think a specific Widget should be used for it and the UI of this Widget should clearly delimit the group, like a Container with a border or something, otherwise the UI will not show anything different... in fact the idea is to support "in the future" some kind of customization for groups... as well as some kind of customization for other type of items by using extensions...